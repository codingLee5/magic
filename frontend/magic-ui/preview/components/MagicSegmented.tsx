import React, { useState } from "react"
import { Space } from "antd"
import MagicSegmented from "../../components/MagicSegmented"
import ComponentDemo from "./Container"

const SegmentedDemo: React.FC = () => {
	const [value1, setValue1] = useState<string>("选项1")
	const [value2, setValue2] = useState<string>("选项1")

	return (
		<div>
			<ComponentDemo
				title="基础分段控制器"
				description="最基本的分段控制器组件"
				code="<MagicSegmented options={['选项1', '选项2', '选项3']} />"
			>
				<Space>
					<MagicSegmented options={["选项1", "选项2", "选项3"]} />
					<MagicSegmented circle={false} options={["选项1", "选项2", "选项3"]} />
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="受控分段控制器"
				description="通过value和onChange控制选中状态"
				code="value | onChange"
			>
				<Space direction="vertical">
					<MagicSegmented
						options={["选项1", "选项2", "选项3"]}
						value={value1}
						onChange={setValue1}
					/>
					<span>当前选中: {value1}</span>
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="不同尺寸"
				description="支持不同尺寸的分段控制器"
				code="size: 'large' | 'default' | 'small'"
			>
				<Space direction="vertical">
					<MagicSegmented options={["选项1", "选项2", "选项3"]} size="large" />
					<MagicSegmented options={["选项1", "选项2", "选项3"]} />
					<MagicSegmented options={["选项1", "选项2", "选项3"]} size="small" />
				</Space>
			</ComponentDemo>

			<ComponentDemo title="禁用状态" description="支持禁用某些选项" code="disabled">
				<Space>
					<MagicSegmented
						options={[
							{ label: "选项1", value: "选项1" },
							{ label: "选项2", value: "选项2", disabled: true },
							{ label: "选项3", value: "选项3" },
						]}
					/>
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="事件处理"
				description="监听选择变化事件"
				code="onChange: (value) => void"
			>
				<Space direction="vertical">
					<MagicSegmented
						options={["选项1", "选项2", "选项3"]}
						value={value2}
						onChange={(value) => {
							setValue2(value)
							console.log("选择变化:", value)
						}}
					/>
					<span>当前选中: {value2}</span>
				</Space>
			</ComponentDemo>
		</div>
	)
}

export default SegmentedDemo
