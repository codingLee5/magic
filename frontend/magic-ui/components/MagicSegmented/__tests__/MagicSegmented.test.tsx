import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect, vi } from "vitest"
import MagicSegmented from "../index"

describe("MagicSegmented", () => {
	it("should render without crashing", () => {
		const options = [
			{ label: "选项1", value: "1" },
			{ label: "选项2", value: "2" },
			{ label: "选项3", value: "3" },
		]

		render(
			<MagicThemeProvider theme="light">
				<MagicSegmented options={options} />
			</MagicThemeProvider>,
		)

		// 检查是否渲染了选项
		expect(screen.getByText("选项1")).toBeInTheDocument()
		expect(screen.getByText("选项2")).toBeInTheDocument()
		expect(screen.getByText("选项3")).toBeInTheDocument()
	})

	it("should handle value change", () => {
		const options = [
			{ label: "选项1", value: "1" },
			{ label: "选项2", value: "2" },
		]
		const onChange = vi.fn()

		render(
			<MagicThemeProvider theme="light">
				<MagicSegmented options={options} onChange={onChange} />
			</MagicThemeProvider>,
		)

		const option2 = screen.getByText("选项2")
		option2.click()

		expect(onChange).toHaveBeenCalledWith("2")
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
				{ label: "选项3", value: "3" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带默认值分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
				{ label: "选项3", value: "3" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} defaultValue="2" />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("禁用状态分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} disabled />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("大尺寸分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} size="large" />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("小尺寸分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} size="small" />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带图标分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1", icon: <span>🚀</span> },
				{ label: "选项2", value: "2", icon: <span>⭐</span> },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("块级分段控制器快照", () => {
			const options = [
				{ label: "选项1", value: "1" },
				{ label: "选项2", value: "2" },
			]

			const { asFragment } = render(
				<MagicThemeProvider theme="light">
					<MagicSegmented options={options} block />
				</MagicThemeProvider>,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
