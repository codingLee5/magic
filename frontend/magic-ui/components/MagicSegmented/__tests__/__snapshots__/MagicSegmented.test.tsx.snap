// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicSegmented > 快照测试 > 块级分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk magic-segmented-block css-dev-only-do-not-override-yotw4n css-var-rg"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 基础分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk css-dev-only-do-not-override-yotw4n css-var-r4"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项3"
        >
          选项3
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 大尺寸分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk magic-segmented-lg css-dev-only-do-not-override-yotw4n css-var-ra"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 小尺寸分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk magic-segmented-sm css-dev-only-do-not-override-yotw4n css-var-rc"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 带图标分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk css-dev-only-do-not-override-yotw4n css-var-re"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
        >
          <span
            class="magic-segmented-item-icon"
          >
            <span>
              🚀
            </span>
          </span>
          <span>
            选项1
          </span>
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
        >
          <span
            class="magic-segmented-item-icon"
          >
            <span>
              ⭐
            </span>
          </span>
          <span>
            选项2
          </span>
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 带默认值分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented acss-1yvjgsk css-dev-only-do-not-override-yotw4n css-var-r6"
    role="radiogroup"
    tabindex="0"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item magic-segmented-item-selected"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
      <label
        class="magic-segmented-item"
      >
        <input
          class="magic-segmented-item-input"
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项3"
        >
          选项3
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSegmented > 快照测试 > 禁用状态分段控制器快照 1`] = `
<DocumentFragment>
  <div
    aria-label="segmented control"
    class="magic-segmented magic-segmented-disabled acss-1yvjgsk css-dev-only-do-not-override-yotw4n css-var-r8"
    role="radiogroup"
  >
    <div
      class="magic-segmented-group"
    >
      <label
        class="magic-segmented-item magic-segmented-item-selected magic-segmented-item-disabled"
      >
        <input
          checked=""
          class="magic-segmented-item-input"
          disabled=""
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="true"
          class="magic-segmented-item-label"
          title="选项1"
        >
          选项1
        </div>
      </label>
      <label
        class="magic-segmented-item magic-segmented-item-disabled"
      >
        <input
          class="magic-segmented-item-input"
          disabled=""
          name="test-id"
          type="radio"
        />
        <div
          aria-selected="false"
          class="magic-segmented-item-label"
          title="选项2"
        >
          选项2
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
